scripts/spelling.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset
+
+check_prerequisites() {
+    if [[ -n ${RD_LINT_SKIP_SPELLING:-} ]]; then
+        echo "Skipping spell checking in CI."
+        exit
+    fi
+
+    case $(uname -s) in # BSD uname doesn't support long option `--kernel-name`
+        Darwin) check_prerequisites_darwin;;
+        Linux) check_prerequisites_linux;;
+        CYGWIN*|MINGW*|MSYS*) check_prerequisites_windows;;
+        *) printf "Prerequisites not checked on %s\n" "$(uname -s)" >&2 ;;
+    esac
+}
+
+check_prerequisites_darwin() {
+    if ! command -v cpanm &>/dev/null; then
+        echo "Please install cpanminus first:" >&2
+        if command -v brew &>/dev/null; then
+            echo "brew install cpanminus" >&2
+        fi
+        exit 1
+    fi
+    # On macOS, the spell checker fails to skip expected long words.
+    # Disable spell checking there until check-spelling releases v0.0.25.
+    # https://github.com/check-spelling/check-spelling/issues/84
+    echo "Skipping spell checking, macOS has false positives."
+    exit
+}
+
+check_prerequisites_linux() {
+    if command -v wslpath >&/dev/null; then
+        check_prerequisites_windows
+        return
+    fi
+    if [[ -z "${PERL5LIB:-}" ]]; then
+        export PERL5LIB=$HOME/perl5/lib/perl5
+    fi
+    if command -v cpanm &>/dev/null; then
+        return
+    fi
+    echo "Please install cpanminus first:" >&2
+    if command -v zypper &>/dev/null; then
+        echo "zypper install perl-App-cpanminus" >&2
+    elif command -v apt &>/dev/null; then
+        echo "apt install cpanminus" >&2
+    fi
+    exit 1
```

Comment:
I understand that you want to avoid adding a nesting level, but I think the Darwin and Linux versions of the function should follow the same logic flow:

```suggestion
    if ! command -v cpanm &>/dev/null; then
      echo "Please install cpanminus first:" >&2
      if command -v zypper &>/dev/null; then
          echo "zypper install perl-App-cpanminus" >&2
      elif command -v apt &>/dev/null; then
          echo "apt install cpanminus" >&2
      fi
      exit 1
    fi
```

Reply:
Once check-spelling v0.0.25 is released (where the darwin bug is fixed), I'll actually make the darwin one look more like the linux one.

I can alternatively make a helper for finding the commands if that looks better, _and_ we don't need to support bash 3 (shipped with macOS):
```bash
cpan_install_commands() {
  declare -A commands=(
    [apt-get]="apt-get install cpanminus"
    [brew]="brew install cpanminus"
    [zypper]="zypper install perl-App-cpanminus"
  )
  echo "Please install cpanminus first:" >&2
  local key
  for key in "${!commands[@]}"; do
    if command -v "$key" &>/dev/null; then
      echo "${commands[$key]}" >&2
    fi
  done
  exit 1
}

check_prerequisites_linux() {
  …
  if ! command -v cpanm &>/dev/null; then
    cpan_install_commands
  fi
}
```

If we want bash 3 support, I can use an `if/elif` chain instead.  No associative arrays back then.

Reply:
Leave it for now until check-spelling 0.25 is released.